internal/raft/raft.go

```diff
@@ -41,12 +41,12 @@ type Node struct {
 type PersistentState struct {
 	CurrentTerm int32
 	VotedFor    id.ID // VotedFor is nil at init, and id.ID of the node after voting is complete.
+	mu          sync.Mutex
```

Comment:
```suggestion

	logLock          sync.Mutex
```
Should probably also be named differently to make more clear what it protects.
Also, new line before is kind of mandatory imo

Reply:
This goes in the `raft` branch, not here. Let this stay like this

Reply:
i will revert it